clang/include/clang/CIR/Dialect/IR/CIRTypes.td

```diff
@@ -94,6 +95,70 @@ def SInt16 : SInt<16>;
 def SInt32 : SInt<32>;
 def SInt64 : SInt<64>;
 
+//===----------------------------------------------------------------------===//
+// FloatType
+//===----------------------------------------------------------------------===//
+
+def CIR_FloatType : CIR_Type<"Float", "float",
```

Comment:
@Lancern @bcardosolopes what do you think of breaking this down so that each floating point type has its own CIR counterpart?

It would remove a lot of bloat like the predicate methods (`isBinary32`, `isBinary64`, etc.), make the print/parse/verify/get methods unnecessary.

The `CIR_AnyFloatType` can be used to join these types on a single constraint and any shared properties may be introduced through traits or interfaces.

Here's an [example of how LLVM Dialect does it](https://github.com/llvm/llvm-project/blob/e22cb93890c33e21534338e4f2ea5ce640c78b77/mlir/include/mlir/IR/BuiltinTypes.td#L68).

Reply:
Yes, @Lancern please address this.